pkg/podmanager/pod_lifecycle_manager.go

```diff
@@ -43,7 +44,7 @@ type PodLifecycleManager interface {
 	DeletePod(context.Context, client.Client, *corev1.Pod) error
 
 	// CanDeletePods checks whether it is safe to delete pods.
-	CanDeletePods(context.Context, fdbadminclient.AdminClient, *fdbv1beta2.FoundationDBCluster) (bool, error)
+	CanDeletePods(context.Context, logr.Logger, fdbadminclient.AdminClient, *fdbv1beta2.FoundationDBCluster) (bool, error)
```

Comment:
We can't change the interface here. That would be a breaking change and requires a new major version and adding a new method will also break the interface in an incompatible way. We can discuss some ideas how we can move forward with this.

Reply:
Like discussed offline we will make use of https://pkg.go.dev/github.com/go-logr/logr#NewContext and https://pkg.go.dev/github.com/go-logr/logr#FromContext